Add SECURITY.md (#4)
hibernate/.github: SECURITY.md

## **Reporting a Vulnerability**

We take the security of our projects seriously. If you discover a security vulnerability,
we appreciate your help in disclosing it to us in a responsible manner.

**Please DO NOT open a public JIRA/GitHub issue.**

Instead, please report the vulnerability directly as described at

> the [Security Contacts and Procedures](https://access.redhat.com/security/team/contact/#contact).

Reviewer: I wonder why RH is the right place to report security issues. @ebullient?

Author: Yeah, I was also second guessing myself, but since it's what we still have on the hibernate.org ...

https://hibernate.org/orm/

See the security issue in the "menu" on the left bottom.

Reviewer: Not claiming it's wrong, but Yoann is on PTO so I can't ask him.

Reviewer 2: I think you're free to define a new place that is not tied to your employer. It could be inbound to project-hibernate@commonhaus.org, or you could create a security@hibernate.org alias.. or you could enable private vulnerability reporting on the repos themselves.

https://docs.github.com/en/code-security/security-advisories/guidance-on-reporting-and-writing-information-about-vulnerabilities/privately-reporting-a-security-vulnerability

Author: thanks Erin 🙂 
I  like the GH option, let's wait for @yrodiere though 😄

Reviewer 2: whatever you do, please also have @yrodiere send it around to other project reps, as they will need something similar.. may as well share. ;)

Reviewer: @ebullient you don't think CF itself should have at least a policy on this?

Reviewer 3: This is not as simple as "let's just set up a @hibernate.org alias", because:

1. Companies with Hibernate-based products will want their security department to be aware of problems ASAP, and IMO that's fair if the company contributes significantly. So that alias should include any such company's security team, not just our team.
2. Some people's reports are simply invalid... and I very much value the work of security experts screening the reports and only passing on to us whatever _actually_ makes sense.

I'd say for now let's stick to the Red Hat address: they're currently the only ones supporting Hibernate, the only ones with contractually binding commitments regarding security, and they'll pass on to us whatever valid reports they get anyway.

I'll add to my todo list "figure out how to handle security reports in our new multi-stakeholder situation".

### **Supported Versions**

> We only provide security updates for the **stable releases**.

Reviewer 3: ```suggestion
We provide security fixes for the **latest stable release** of each project.
```

> We only provide security updates for the **latest stable release** and releases that are in the limited-support state.

Reviewer 3: Is this extracted from somewhere else? Because in my mind "limited-support" meant that we _might_ provide security updates, but don't provide guarantees. It's entirely possible the update would be provided for "supported builds" only (SP, ...).

Author: This was just my understanding of those "limited-support-versions" e.g. current stable Hibernate Search version is 8.0 and we have 7.2 7.1 6.2 and 5.10 in limited support, so the way I was "reading" it was that we'd only work on bug fixes for 8.0 but if there's something critical (e.g. a cve) then we'd backport it to those other versions. But maybe I just misinterpreted it 🫣 😃

Reviewer 3: I think you're expecting too much. Your understanding may be correct in practice, but it's not what we _promise_.

See https://hibernate.org/community/maintenance-policy/

Reviewer 3: In particular:

> Backports to these series are highly restricted and releases are not guaranteed.

> Bug-fixes and improvements are only applied to latest-stable/stable and development (and sometimes to limited-support) series.

> Only development and latest-stable/stable series are actively released. Occasional releases of limited-support series may happen, but should not be expected.

> To find out which versions match this criteria go to the corresponding project page on https://hibernate.org/.

Reviewer 3: ```suggestion
To find out which versions match this criteria go to the corresponding project page on [our website](https://hibernate.org/). For example:

* [Hibernate ORM](https://hibernate.org/orm/releases/)
* [Hibernate Search](https://hibernate.org/search/releases/)
* [Hibernate Validator](https://hibernate.org/validator/releases/)
* [Hibernate Reactive](https://hibernate.org/reactive/releases/)
```

> It is recommended to use the most recent version to ensure you have the latest security patches and bug fixes.

Reviewer 3: ```suggestion
We recommend you use the most recent version to ensure you have the latest security patches and bug fixes.
```

We may fix the vulnerability in older versions depending on the severity of the issue and the age of the release,

> but we are only committing to the versions described above.

Reviewer 3: I'd skip this and reword above to avoid being redundant.

```suggestion
```
